src/components/editor/components/RecipientInput.js

```diff
@@ -3,246 +3,219 @@
+// `WithContext` as `ReactTags` is taken from the official example
+import {WithContext as ReactTags} from 'react-tag-input';
 import * as l10n from '../../../lib/l10n';
+import {checkEmail, getUUID} from '../../../lib/util';
 
 import './RecipientInput.scss';
 
-/* global angular */
-
 l10n.register([
   'editor_key_has_extra_msg',
   'editor_key_not_found',
   'editor_key_not_found_msg'
 ]);
 
-/*
-  reference to props of RecipientInput will be shared with Angular controller
-  this structure is not immutable, recipients will be received as {email},
-  but RecipientInputCtrl will modify recipients to {email, keys}
+/**
+ * Checks if all recipients have a public key and prevents encryption
+ * if one of them does not have a key.
+ * @param {Tag[]} tags - Array of tags
+ * @param {Key[]} keys - Array of public keys
+ * @param {Boolean} extraKey - Flag indicating whether extra key input is enabled
+ * @returns {Boolean} true if any recipient is unencrypted
  */
-const contrCompStack = [];
-
-export class RecipientInput extends React.Component {
-  constructor(props) {
-    super(props);
-    this.id = getUUID();
-  }
-
-  propsOnStack() {
-    // store props on stack for Angular
-    this.ctrlLink = {props: this.props};
-    contrCompStack.push(this.ctrlLink);
-  }
-
-  componentDidMount() {
-    // load editor module dependencies
-    angular.module('recipientInput', ['ngTagsInput'])
-    .config((tagsInputConfigProvider, $locationProvider) => {
-      // activate monitoring of placeholder option
-      tagsInputConfigProvider.setActiveInterpolation('tagsInput', {placeholder: true});
-      $locationProvider.hashPrefix('');
-    });
-    // attach ctrl to editor module
-    angular.module('recipientInput').controller('RecipientInputCtrl', RecipientInputCtrl);
-    this.propsOnStack();
-    // bootstrap angular
-    angular.bootstrap(document.getElementById(this.id), ['recipientInput']);
-    if (this.ctrlLink.props.recipients.length) {
-      this.ctrlLink.rInputCtrl.recipients = this.ctrlLink.props.recipients;
-      this.ctrlLink.rInputCtrl.update();
-    }
-  }
-
-  shouldComponentUpdate(nextProps) {
-    this.ctrlLink.props = nextProps;
-    this.ctrlLink.rInputCtrl.recipients = this.ctrlLink.props.recipients;
-    // only update input controller if recipients or keys change
-    if (this.props.recipients !== nextProps.recipients ||
-        this.props.keys !== nextProps.keys ||
-        this.props.extraKey !== nextProps.extraKey) {
-      this.ctrlLink.rInputCtrl.update();
+function isAnyRecipientUnencrypted(tags, keys, extraKey) {
+  return tags.some(t => !findRecipientKey(keys, t.id)) && !extraKey;
+}
+
+/**
+ * Creates a bootstrap badge class name
+ * @param {Boolean} isSuccess - Flag indicating whether the recipient has a key
+ * @param {Boolean} hasExtraKey - Flag indicating whether extra key input is enabled
+ * @returns {String} CSS class name for the tag
+ */
+const getTagClassName = (isSuccess, hasExtraKey) =>
```

Comment:
Unclear why this is no function declaration. If no reason, then should be equalized with the other function declarations in this scope.